lms/services/moodle.py

```diff
@@ -99,10 +107,20 @@ def list_files(self, course_id: int):
         )
         return file_tree
 
+    def file_exists(self, file_id) -> bool:
+        """Check if the file exists in the course."""
```

Comment:
There's a few warts in this PR because the API behaviors a bit wirdly. Doing a HEAD request here is the best compromise I could find.